core/src/main/scala/libretto/CoreLib.scala

```diff
@@ -316,11 +314,11 @@ class CoreLib[DSL <: CoreDSL](val dsl: DSL) { lib =>
       introFst(lInvertPongPing > swap) > assocLR > snd(notifyResumeNeg)
 
     /** Signals resumption. */
-    implicit def signalingDeferred[A]: Signaling.Negative[Deferred[A]] =
+    given signalingDeferred[A]: Signaling.Negative[Deferred[A]] =
       Signaling.Negative.byFst
 
     /** Defers resumption. */
-    implicit def deferrableDeferred[A]: Deferrable.Negative[Deferred[A]] =
+    given deferrableDeferred[A]: Deferrable.Negative[Deferred[A]] =
```

Comment:
This is problematic. The names are just a repetition of what the types say already. So I would prefer not to have names for the givens here.  But Scala seems to have issues. Both givens get the same name synthesized… So the name is needed.

I think opening a compiler issue would be the right way to handle this.

Reply:
Yeah, let's keep them named for now, maybe report issue and resolve later.